src/Library/demos/Avatar/code.rs

```diff
@@ -0,0 +1,38 @@
+use crate::workbench;
+use adw::prelude::*;
+use gtk::{gdk, gio, glib};
+
+pub fn main() {
+    gtk::init().unwrap();
+
+    let button: gtk::Button = workbench::builder().object("button").unwrap();
+
+    let mut filter = gtk::FileFilter::new();
+    filter.set_name(Some("Images"));
+    filter.add_pixbuf_formats();
+
+    let dialog = gtk::FileDialog::builder()
+        .title("Select an Avatar")
+        .modal(true)
+        .default_filter(&filter)
+        .build();
+
+    button.connect_clicked(move |_| {
+        dialog.open(
+            Some(workbench::window()),
+            None::<&gio::Cancellable>,
+            on_selected,
+        )
+    });
+}
+
+fn on_selected(result: Result<gio::File, glib::Error>) {
```

Comment:
```suggestion
    button.connect_clicked(move |_| {
        dialog.open(
            Some(workbench::window()),
            None::<&gio::Cancellable>,
            on_clicked,
        )
    });
}

fn on_clicked(result: Result<gio::File, glib::Error>) {
```

To use the same naming as the other demos